test/OpenTelemetry.AotCompatibility.Tests/AotCompatibilityTests.cs

```diff
@@ -85,7 +85,7 @@ public void EnsureAotCompatibility()
             Assert.True(process.ExitCode == 0, "Publishing the AotCompatibility app failed. See test output for more details.");
 
             var warnings = expectedOutput.ToString().Split('\n', '\r').Where(line => line.Contains("warning IL"));
-            Assert.Equal(36, warnings.Count());
+            Assert.Equal(42, warnings.Count());
```

Comment:
Interesting this is going up with these changes. I would expect it to be going down always. Should we add suppressions in the test for where we know this code is being called?

Reply:
This is typical for trim/AOT warnings. If you have a centralized method that is doing something incompatible, and you annotate it as such, now you start getting warnings for all the callsites of that method, and you need to fix them.

Reply:
@CodeBlanch:
I originally was thinking to isolate changes of the callers into the PropertyFetcher, and create follow-up PRs for that.

But SqlClient code changes are very straightforward; I just pushed a commit for that:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4675/commits/4a7534cb642930d8a3e62e140c7727bf802c445f

Reply:
@Yun-Ting Could you move the Sql changes to a separate PR? It would be to see its impact alone on the number of warnings instead of mixing up with PropertyFetcher's effect on the warnings. (as PropertyFetcher had an increasing effect on it)

Reply:
I've reverted the SqlClientInstrumentation suppression changes.